api/app.ts

```diff
@@ -1227,30 +1244,86 @@ app.post(
 						const editableOptions: any = {}
-						const editableZWaveProps = [
-							'attempts',
-							'disableOptimisticValueUpdate',
-							'emitValueUpdateAfterSetValue',
-							'inclusionUserCallbacks',
-							'joinNetworkUserCallbacks',
-							'interview',
-							'logConfig',
-							'preferences',
-							'vendor',
-							'userAgent',
-						]
-
-						// Only include properties that actually changed
-						for (const key of editableZWaveProps) {
-							if (
-								!utils.deepEqual(
-									actualSettings.zwave?.[key],
-									settings.zwave?.[key],
-								) &&
-								settings.zwave?.[key] !== undefined
-							) {
-								editableOptions[key] = settings.zwave[key]
+
+						// Check disableOptimisticValueUpdate
+						if (
+							!utils.deepEqual(
+								actualSettings.zwave
+									?.disableOptimisticValueUpdate,
+								settings.zwave?.disableOptimisticValueUpdate,
+							) &&
+							settings.zwave?.disableOptimisticValueUpdate !==
+								undefined
+						) {
+							editableOptions.disableOptimisticValueUpdate =
+								settings.zwave.disableOptimisticValueUpdate
+						}
+
+						// Check scales (maps to preferences.scales)
+						if (
+							!utils.deepEqual(
+								actualSettings.zwave?.scales,
+								settings.zwave?.scales,
+							)
+						) {
+							const preferences = convertScalesToPreferences(
+								settings.zwave?.scales,
+							)
+							if (preferences) {
+								editableOptions.preferences = preferences
+							}
+						}
+
+						// Check logConfig properties
+						const logConfigChanged =
+							!utils.deepEqual(
+								actualSettings.zwave?.logEnabled,
+								settings.zwave?.logEnabled,
+							) ||
+							!utils.deepEqual(
+								actualSettings.zwave?.logLevel,
+								settings.zwave?.logLevel,
+							) ||
+							!utils.deepEqual(
+								actualSettings.zwave?.logToFile,
+								settings.zwave?.logToFile,
+							) ||
+							!utils.deepEqual(
+								actualSettings.zwave?.maxFiles,
+								settings.zwave?.maxFiles,
+							) ||
+							!utils.deepEqual(
+								actualSettings.zwave?.nodeFilter,
+								settings.zwave?.nodeFilter,
+							)
+
+						if (logConfigChanged) {
+							// Build logConfig object from our settings
+							const loglevels: Record<string, any> = {
+								error: 'error',
+								warn: 'warn',
+								info: 'info',
+								verbose: 'verbose',
+								debug: 'debug',
+								silly: 'silly',
+							}
+
+							editableOptions.logConfig = {
+								enabled: settings.zwave?.logEnabled,
+								level: settings.zwave?.logLevel
+									? loglevels[settings.zwave.logLevel]
+									: 'info',
+								logToFile: settings.zwave?.logToFile,
+								maxFiles: settings.zwave?.maxFiles || 7,
+								nodeFilter:
+									settings.zwave?.nodeFilter &&
+									settings.zwave.nodeFilter.length > 0
+										? settings.zwave.nodeFilter.map(
+												(n: string) => parseInt(n),
+											)
+										: undefined,
```

Comment:
like with scales move this to a function that returns logConfig and reuse it on zwaveclient connect